include/mapbox/cheap_ruler.hpp

```diff
@@ -134,6 +134,10 @@ class CheapRuler {
     double lineDistance(const line_string& points) {
         double total = 0.;
 
+        if (points.empty()) {
+            return total;
+        }
+
         for (unsigned i = 0; i < points.size() - 1; ++i) {
```

Comment:
Is the early return like this necessary? The loop should be noop in case of empty vector (`size()` will be 0), so it will work regardless, no?

Reply:
Not quite - a segmentation fault occurs because `i` is unsigned, and given `size` is 0, `0 - 1` evaluates to `UINT_MAX`, causing the loop to occur, and thus accessing `points` entries that don't exist.

Reply:
Would it be simpler to make `i` signed, or is that an antipattern?

Reply:
[mapbox::geometry::line_string](https://github.com/mapbox/geometry.hpp/blob/master/include/mapbox/geometry/line_string.hpp) is a generic container that uses [std::vector](http://en.cppreference.com/w/cpp/container/vector) as default - which uses an unsigned integer as default `size_type`. I suppose the rationale is that a container cannot have a negative amount of items.

Reply:
I've made a PR in geometry.hpp - https://github.com/mapbox/geometry.hpp/pull/57 - that exposes the geometry's underlying size type via `::size_type` alias.